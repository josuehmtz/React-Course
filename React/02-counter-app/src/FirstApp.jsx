// //?Esta es una opciÃ³n de como se puede poner varios elementos dentro del functional component
// import { Fragment } from "react"

// export const FirstApp = () => {
//   return (
//     <Fragment>
//       <div>Josue Hiram</div>
//       <div>Parent element example</div>
//     </Fragment>

//   )
// }


const getName = () => 'Hello Hiram';

//* If the value won't change, the constant should be outside of the functional component

const name = 'Hello Josue, welcome to your first app';

const newMessage = {
  message: 'Hello World', 
  name : 'Josue Hiram'
}

export const FirstApp = () => {
  return (
    <>
      <h1>{ getName() }</h1>
      {/* <h1>{ name }</h1> */}
      {/* <code>{ JSON.stringify(newMessage) }</code> */}
      <p>Parent element example</p>
    </>

  )
}
