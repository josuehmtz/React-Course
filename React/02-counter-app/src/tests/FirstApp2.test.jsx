import { getByText, render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";


//? screen es el componente renderizado

//Validate if a component contains a title

//Snapshot testing debería de ser cuando np vaya a ver muchos cambios

describe('Pruebas en FirstApp', () => {

    const title    = 'FirstApp Test';
    const subtitle = 'Subititle test';

    test('debe de hacer match con el container', () => { 
        const { container } = render(
            <FirstApp title={ title }/>
        );

        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje First App Test', () => { 
        render(
            <FirstApp title={ title }/>
        )
        expect( screen.getByText(title)).toBeTruthy();
        //* para mostrar el componente renderizado screen.debug();
    })

    test('Debe de mostrar el título en un h1', () => {
        render(
            <FirstApp title={title}/>
        );

        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain( title );
    })

    // test('Debe de mandar el subititulo mandado por props', () => { 
    //     render(
    //         <FirstApp subtitle={subititle}/>
    //     )

    //     expect(screen.getAllByAltText( subtitle ).length).toBe( 2 );
    // })
});