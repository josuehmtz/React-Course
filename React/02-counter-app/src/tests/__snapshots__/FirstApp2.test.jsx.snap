// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe de hacer match con el container 1`] = `
<div>
  <h1
    data-testid="testTitle"
  >
    FirstApp Test
  </h1>
  <p>
    Parent element example
  </p>
</div>
`;
